objcgen/objcgenerator.cs

```diff
@@ -599,14 +602,21 @@ void ImplementMethod (MethodInfo info, string name, bool isExtension = false, Pr
 			builder.EndImplementation ();
 		}
 
-		protected override void Generate (MethodInfo mi)
+		protected override void Generate (ProcessedMethod method)
 		{
+			MethodInfo mi = method.Method;
 			string name;
+
 			if (mi.IsSpecialName && mi.IsStatic && mi.Name.StartsWith ("op_", StringComparison.Ordinal))
 				name = CamelCase (mi.Name.Substring (3));
 			else
 				name = CamelCase (mi.Name);
```

Comment:
That logic needs to be moved into the processor as this renaming can cause duplicates, e.g. `op_Addition` would become `Addition` but there could already be a method using that name.

note: does not mean it has to be inside this PR - a lot of other moves are needed to ensure uniqueness